Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py

```diff
@@ -173,112 +499,164 @@ def decode_str(decrypted_text: bytes, encoding: str) -> Tuple[str, str]:
+        elif email_type == 'text':
+            msg_out = email_body
+            human_readable = f'### {msg}The decrypted message is: \n ___ \n {msg_out}'
+        else:
+            human_readable = f'### {msg}The decrypted message is signed, verify to get the original message\n ___ \n{out}'
+
+    results = [CommandResults(
+        readable_output=human_readable,
+        outputs_prefix='SMIME.Decrypted',
+        outputs={'Message': msg_out},
+        tags=tags,
+    )]
+
+    if email_type == 'html':
+        results.append(CommandResults(
+            raw_response=msg_out,
+            content_format=EntryFormat.HTML,
+            entry_type=EntryType.NOTE,
+        ))
+
+    return results
+
+
+def sign_and_encrypt(client: Client, args: dict):
+    message = args.get('message', '')
+    sign = argToBoolean(args.get('signed', 'true'))
+    encrypt = argToBoolean(args.get('encrypted', 'true'))
+    sender = args.get('sender', '')
+    subject = args.get('subject', '')
+    create_file = argToBoolean(args.get('create_file_p7', 'false'))
+    attachment_ids = argToList(args.get('attachment_entry_id', ''))  # type: list[str]
 
-    return human_readable, entry_context
+    recipients = safe_load_json(args.get('recipients', {}) or '{}')
+    if not isinstance(recipients, dict):
+        raise DemistoException('Failed to parse recipients. (format `{"recipient@email":"cert", "other@email":"cert"}`)')
 
+    cc = safe_load_json(args.get('cc', {}) or '{}')
+    if not isinstance(cc, dict):
+        raise DemistoException('Failed to parse cc. (format `{"recipient@email":"cert", "other@email":"cert"}`)')
 
-def sign_and_encrypt(client: Client, args: Dict):
+    bcc = safe_load_json(args.get('bcc', {}) or '{}')
+    if not isinstance(bcc, dict):
+        raise DemistoException('Failed to parse bcc. (format `{"recipient@email":"cert", "other@email":"cert"}`)')
 
-    message = args.get('message', '').encode('utf-8')
-    msg_bio = BIO.MemoryBuffer(message)
-    sign = client.private_key_file
-    encrypt = client.public_key_file
+    msg_str = create_message_with_attachments(message, attachment_ids)
+    demisto.debug(f'\n\nMessage:\n\n {msg_str} \n\nMessage end\n')
 
     if sign:
+        msg_bio = BIO.MemoryBuffer(msg_str.encode('utf-8'))
         client.smime.load_key(client.private_key_file, client.public_key_file)
-        if encrypt:
-            p7 = client.smime.sign(msg_bio, flags=SMIME.PKCS7_TEXT)
-        else:
-            p7 = client.smime.sign(msg_bio, flags=SMIME.PKCS7_TEXT | SMIME.PKCS7_DETACHED)
-        msg_bio = BIO.MemoryBuffer(message)  # Recreate coz sign() has consumed it.
+        p7 = client.smime.sign(msg_bio, algo='sha256')
 
     if encrypt:
-        x509 = X509.load_cert(client.public_key_file)
-        sk = X509.X509_Stack()
-        sk.push(x509)
-        client.smime.set_x509_stack(sk)
+        pub_certs = [cert for dest in [recipients, cc, bcc] for cert in dest.values()]  # all keys are used the same
```

Comment:
I don't understand the comment `# all keys are used the same`, what does it mean?

Reply:
I'll rephrase it.
We just consolidate all recipient certificates for encryption